packages/client/hmi-client/src/components/workflow/tera-create-workflow-modal.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+	<tera-modal class="w-9">
+		<template #header>
+			<h4>Create a new workflow</h4>
+		</template>
+		<template #default>
+			<div class="grid">
+				<aside class="col-2">
+					<label>Select a template</label>
+					<div v-for="scenario in scenarios" :key="scenario.template.templateId" class="flex align-items-center py-1">
+						<RadioButton
+							:inputId="scenario.template.templateId"
+							:value="scenario.template.templateId"
+							v-model="selectedTemplateId"
+							:disabled="scenario.template.templateId !== 'blank-canvas'"
+						/>
+						<label class="pl-2" :for="scenario.template.templateId">{{ scenario.template.templateName }}</label>
+					</div>
+				</aside>
+				<main class="col-10 flex flex-column gap-3">
+					<component
+						v-if="getScenario()"
+						:is="getScenario()?.component"
+						:state="getScenario()?.template"
+						@update-state="onUpdateState"
+					/>
+				</main>
+			</div>
+		</template>
+		<template #footer>
+			<Button label="Create" size="large" @click="saveWorkflow" />
+			<Button label="Close" class="p-button-secondary" size="large" outlined @click="emit('close-modal')" />
+		</template>
+	</tera-modal>
+</template>
+
+<script setup lang="ts">
+import TeraModal from '@/components/widgets/tera-modal.vue';
+import Button from 'primevue/button';
+import { ref } from 'vue';
+import type { Component } from 'vue';
+import RadioButton from 'primevue/radiobutton';
+import { BaseScenarioTemplate } from '@/components/workflow/scenario-templates/scenario-template';
+import { createWorkflow } from '@/services/workflow';
+import { AssetType } from '@/types/Types';
+import { useProjects } from '@/composables/project';
+import router from '@/router';
+import { RouteName } from '@/router/routes';
+import TeraBlankCanvasTemplate from '@/components/workflow/scenario-templates/blank-canvas/tera-blank-canvas-template.vue';
+import TeraSituationalAwarenessTemplate from '@/components/workflow/scenario-templates/situational-awareness/tera-situational-awareness-template.vue';
+import { BlankCanvasScenarioTemplate } from '@/components/workflow/scenario-templates/blank-canvas/blank-canvas-template';
+import { SituationalAwarenessScenarioTemplate } from '@/components/workflow/scenario-templates/situational-awareness/situational-awareness-template';
+
+interface ScenarioItem {
+	template: BaseScenarioTemplate;
+	component: Component;
+}
+
+const scenarios = ref<ScenarioItem[]>([
+	{
+		template: new BlankCanvasScenarioTemplate(),
+		component: TeraBlankCanvasTemplate
```

Comment:
There is something a bit odd here, you are doing this in the class file

```
export const BlankCanvasTemplate = new BlankCanvasScenarioTemplate();
```

But you are doing this here:
```
new BlankCanvasScenarioTemplate(),
```

So probably need to get the exports straightened out, I am presuming we do not want to export an instantiation of the scenario.


Also maybe we should drop the "Template" suffix ... not sure if it is really necessary and the names are getting quite long 🤔

Reply:
Ya was was using that exported instance for some experimental while making this and forgot to delete them, they're deleted now.  And I've removed the Template suffix 😄